### PR TITLE
Added tree.go

### PR DESCRIPTION
**Pull request by GOVINDDIXIT**

### CODE REVIEW

File: DataStructures/tree.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// Go's concurrency primitives make it easy to
```
Comment (gauthamzz) [inline DataStructures/tree.go]: do you know what this means or did you copy paste this 😆

Comment (gauthamzz) [reply DataStructures/tree.go]: only be merged,if you explain what this line means 😸

Comment (mubaris) [reply DataStructures/tree.go]: @aswinzz @Anupam-Dagar

Comment (GOVINDDIXIT) [reply DataStructures/tree.go]: Concurrency is the composition of independently executing computations.

Comment (gauthamzz) [reply DataStructures/tree.go]: will give you 50 points,if you write a 200 words + blog post on concurrency in go with images,example etc. 

and put on medium or somewhere 😄 
**only for you**

**Pull request merged**